bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/rewriter/gemini/QueryVisitorRewriter.java

```diff
@@ -0,0 +1,91 @@
+package com.google.zetasql.toolkit.antipattern.rewriter.gemini;
+
+import com.google.zetasql.SqlException;
+import com.google.zetasql.toolkit.antipattern.AntiPatternVisitor;
+import com.google.zetasql.toolkit.antipattern.cmd.InputQuery;
+import com.google.zetasql.toolkit.antipattern.rewriter.prompt.PromptYamlReader;
+import com.google.zetasql.toolkit.antipattern.util.AntiPatternHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class QueryVisitorRewriter {
+    private static final Logger logger = LoggerFactory.getLogger(QueryVisitorRewriter.class);
+    private final PromptYamlReader promptYamlReader;
+    private final AntiPatternHelper antiPatternHelper;
+
+    public QueryVisitorRewriter(AntiPatternHelper antiPatternHelper) throws IOException {
+        this.promptYamlReader = new PromptYamlReader();
+        this.antiPatternHelper = antiPatternHelper;
+    }
+
+    public String rewriteSQL(String inputQuery,
+                             AntiPatternVisitor visitor,
+                             Integer TTL) throws Exception {
+        TTL--;
+        if (TTL < 0) {
+            throw new Exception("LLM couldn't solve the specific anti pattern");
```

Comment:
Could we define a specific Exception class, something like LLMCouldNotRewriteException or TTLExpiredDuringRewriteException ... I'll leave the naming up to you

Reply:
Done